sos/report/plugins/openstack_ironic.py

```diff
@@ -74,10 +122,28 @@ def setup(self):
             self.add_cmd_output("openstack baremetal port list --long")
             self.add_cmd_output("openstack baremetal port group list --long")
 
+        if in_container:
+            for container_name in ['ironic_inspector_dnsmasq',
+                                   'ironic_inspector', 'ironic_pxe_http',
+                                   'ironic_pxe_tftp', 'ironic_neutron_agent',
+                                   'ironic_conductor', 'ironic_api']:
+                self.add_cmd_output('%s exec %s rpm -qa'
+                                    % (self.container_runtime, container_name))
+
     def apply_regex_sub(self, regexp, subst):
         for conf in self.conf_list:
             self.do_path_regex_sub(conf, regexp, subst)
 
+    def running_in_container(self):
+        for runtime in ["docker", "podman"]:
+            container_status = self.exec_cmd(runtime + " ps")
+            if container_status['status'] == 0:
+                for line in container_status['output'].splitlines():
+                    if line.endswith("ironic_api"):
+                        self.container_runtime = runtime
+                        return True
+        return False
```

Comment:
There are methods in `Plugin` for detecting containerized workloads. This can all be done away with and replaced with

```python
in_container = self.container_exists('ironic_api')
```

Reply:
That's even better, thanks!

Reply:
Also, regarding the `RedHatPlugin` comment, what I meant is that, since they were at the bottom of the plugin, they were probably missed by the person that added the containerized config at the top. So moving them at the top will help the next time we need to add/change the paths. I'll clarify the commit message.